src/libraries/System.Formats.Tar/tests/TarEntry/PaxTarEntry.Tests.cs

```diff
@@ -36,6 +35,28 @@ public void Constructor_UnsupportedEntryTypes()
             Assert.Throws<ArgumentException>(() => new PaxTarEntry(TarEntryType.GlobalExtendedAttributes, InitialEntryName));
         }
 
+
+        [Theory]
+        [InlineData("\n", "value")]
+        [InlineData("=", "value")]
+        [InlineData("key", "\n")]
+        [InlineData("\nkey", "value")]
+        [InlineData("k\ney", "value")]
+        [InlineData("key\n", "value")]
+        [InlineData("=key", "value")]
+        [InlineData("ke=y", "value")]
+        [InlineData("key=", "value")]
```

Comment:
```suggestion
        [InlineData("key=", "value")]
        [InlineData("key=", "value\n")]
```

Reply:
The suggested new case (line 49) won't have a different behavior than the original (line 48). The key is checked first for a `=`, and since it's found, it throws. We don't get to check the value.

The last InlineData case (line 51) will verify the ending newline in the value string. And to get that string evaluated, the key had to be evaluated as valid.

Reply:
Not sure I follow. Maybe I'm missing something?

Before applying the `IndexOfAny` suggestion above, the condition in line 118 finds a `=` in the key. Since it's found, we, we throw.

After applying the suggestion, the behavior would be the same: First look for the `=` in the key, and if found, throw.

Neither would continue to verify the value string.